libs/util/code-gen-types/src/dotnet-plugin-events-params.types.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  Entity,
+  EntityLookupField,
+  ModuleContainer,
+  ModuleAction,
+} from "./code-gen-types";
+import { IFile } from "./files";
+import { EventParams } from "./plugins.types";
+import { AstNode } from "@amplication/csharp-ast";
+
+export interface CreateEntityServiceBaseParams extends EventParams {
+  entity: Entity;
+  resourceName: string;
+  apisDir: string;
+  file: IFile<AstNode>;
```

Comment:
```suggestion
  file: IFile<AstNode>;
```

AstNode is a great solution when we want a super generic solution but passing a more "expected" type like CSharp file that can be of type `Interface`, `Class` or `Enum` will make plugin experience much easier as it would be possible to get properties by name.

i.e.
```cs 
// IFile<Class> file will allow us to extend easily the obj as follow

const createMethod = file.getMethod("createX");

createMethod.AddAnnotation({ /.../ });

//...
```

Reply:
Outdated. We will have types per event that will improve devex